libpod/runtime.go

```diff
@@ -677,6 +677,53 @@ func (r *Runtime) GetConfig() (*config.Config, error) {
 	return config, nil
 }
 
+// libimageEvents spawns a goroutine in the background which is listenting for
+// events on the libimage.Runtime.  The gourtine will be cleaned up implicitly
+// when the main() exists.
+func (r *Runtime) libimageEvents() {
+	toLibpodEventStatus := func(e *libimage.Event) events.Status {
+		switch e.Type {
+		case libimage.EventTypeImagePull:
+			return events.Pull
+		case libimage.EventTypeImagePush:
+			return events.Push
+		case libimage.EventTypeImageRemove:
+			return events.Remove
+		case libimage.EventTypeImageLoad:
+			return events.LoadFromArchive
+		case libimage.EventTypeImageSave:
+			return events.Save
+		case libimage.EventTypeImageTag:
+			return events.Tag
+		case libimage.EventTypeImageUntag:
+			return events.Untag
+		case libimage.EventTypeImageMount:
+			return events.Mount
+		case libimage.EventTypeImageUnmount:
+			return events.Unmount
+		default:
+			return "Unknown"
+		}
+	}
```

Comment:
A couple of questions.  If this is going to hang around for a while and be used multiple times, might it be better to create a map and reference that?  If you decide to stick with the switch, are these in the order that you think most likely to be found/used?  If the ordering isn't a concern for perf reasons, then can you alphabetize them?

Reply:
Great idea using a map!  I think alphabetic sorting makes sense if there's no relation among the items.  In that case, I prefer tag/untag or mount/unmout, for instance, to stick together.  But I don't feel strongly about it.